### PR TITLE
Improve and simplify AutoEval

### PR DESCRIPTION
**Pull request by joaquimgomez**
Once applied this PR will fix typos and errors, and will improve and simplify the package:

- Update FLIP and biotrainer versions
- Fix README to be simple and accurate with the current features
- Solve incompatibility between `embedder_name` and `embeddigns_file` (mutual exclusion)
- Delete CSV to FASTA conversion since is not required anymore.
- Include Dockerfile
- Allow the mask file input to be arbitrary.
- Update configuration files according to the last included features in biotrainer.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#docker pull ghcr.io/sacdallago/biotrainer:latest
```
Comment (SebieF) [inline Dockerfile:1]: This line can be removed

File: README.md [hunk 2]
```diff
@@ -80,30 +94,21 @@ The available input parameters are:
 
 Availability semaphore:
 - ✅: Available in configsbank in the closest possible way to the better configuration in the reference.
```
Comment (SebieF) [inline README.md:96]:
```suggestion
- ✅: Available in configsbank is the closest possible way to the best configuration in the reference.
```

File: autoeval/managers/data.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def filter_fasta_entries(fasta_entries: List[any], min_size: int, max_size: int)
 
     new_fasta_entries = []
     for entry in fasta_entries:
-        if (min_size is not None and len(entry.seq) > min_size) or (max_size is not None and len(entry.seq) < max_size):
+        if (min_size is not None and len(entry.seq) > min_size) 222 (max_size is not None and len(entry.seq) < max_size):
```
Comment (SebieF) [inline autoeval/managers/data.py:28]:
```suggestion
        if (min_size is not None and len(entry.seq) > min_size) or (max_size is not None and len(entry.seq) < max_size):
```

**Review by SebieF — CHANGES_REQUESTED** [review]
Nice :) Besides the minor issues, I am wondering if autoeval is able to identify the correct fasta file from FLIP directly if only the split is provided as an argument? If this is not the case I would suggest to add this functionality for convenience, but maybe I am also just missing something..

Another nice thing to do would be to update the poetry.lock file via `poetry update` or `poetry lock`. `setuptools = "58.5.1"` can also be removed from the .toml file because it has been updated in biotrainer.

File: autoeval/managers/data.py (hunk 3 above)
Comment (joaquimgomez) [reply autoeval/managers/data.py:28]: I don't know when this happened, but I know I solved it, and I also don't know how this ends here in the repo. Git mysteries.

**Comment by joaquimgomez** [comment]
Thanks @SebieF  for reviewing the PR.

Regarding using AutoEval only with the split name as input argument, the answer is no, it's not possible.
Because, the point of using also as input the protocol and the working directory is that the user can decide to change which protocol they want to use, something feasible for some tasks like SCL (could work with residues/residue_to_class).

But I agree that this could also be simplified/improved by:
- If no working directory is given, then use `./output`
- If no protocol is given, then use the default one defined in `utilities/settings.py.`

If you agree, I implement the idea.

I updated poetry.

**Comment by SebieF** [comment]
Ah sure, making the protocol a required option is fine! Adding the default working directory would be neat :)

I added some typo, format and type fixes to data.py and FASTA.py that I occurred during the review, shouldn't have changed anything about the logic

**Pull request merged**